src/System.Security.Cryptography.OpenSsl/tests/EcDsaOpenSslTests.cs

```diff
@@ -11,46 +11,22 @@ namespace System.Security.Cryptography.EcDsaOpenSsl.Tests
 {
     public static class EcDsaOpenSslTests
     {
-        // On CentOS, secp224r1 appears to be disabled. To prevent test failures on that platform, probe for this capability
-        // before depending on it. 
+        // On CentOS, secp224r1 (also called nistP224) appears to be disabled. To prevent test failures on that platform, 
+        // probe for this capability before depending on it. 
         private static bool ECDsa224Available
         {
             get
             {
-                using (ECDsaOpenSsl key = new ECDsaOpenSsl())
+                try
                 {
-                    KeySizes[] legalKeySizes = key.LegalKeySizes;
-                    return IsLegalSize(224, legalKeySizes);
+                    Oid oid = Oid.FromOidValue("1.3.132.0.33", OidGroup.PublicKeyAlgorithm);
                 }
-            }
-        }
-
-        private static bool IsLegalSize(int size, KeySizes[] legalSizes)
-        {
-            for (int i = 0; i < legalSizes.Length; i++)
-            {
-                KeySizes currentSizes = legalSizes[i];
-
-                // If a cipher has only one valid key size, MinSize == MaxSize and SkipSize will be 0
-                if (currentSizes.SkipSize == 0)
-                {
-                    if (currentSizes.MinSize == size)
-                        return true;
-                }
-                else if (size >= currentSizes.MinSize && size <= currentSizes.MaxSize)
+                catch (CryptographicException)
                 {
-                    // If the number is in range, check to see if it's a legal increment above MinSize
-                    int delta = size - currentSizes.MinSize;
-
-                    // While it would be unusual to see KeySizes { 10, 20, 5 } and { 11, 14, 1 }, it could happen.
-                    // So don't return false just because this one doesn't match.
-                    if (delta % currentSizes.SkipSize == 0)
-                    {
-                        return true;
-                    }
+                    return false;
```

Comment:
I don't know that this is going to be a fully stable approach.  There's a set of OID->name mappings that we maintain as hard-coded to avoid cross-platform confusion.  So things like the Brainpool curves will probably successfully resolve, but still not be supported on CentOS.

https://github.com/dotnet/corefx/blob/master/src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.cs